tests/qactivation_test.py

```diff
@@ -245,6 +247,49 @@ def ref_hard_sigmoid(y):
   expected = sigmoid(test_values)
   assert_allclose(result, expected, rtol=1e-05)
 
+@pytest.mark.parametrize(
+    'bits, sigmoid_type, test_values, expected_values', [
+        (
+            6,
+            "hard",
+            np.array(
+                [[-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75]],
+                dtype=K.floatx()),
+            np.array([[0.   , 0.125, 0.25 , 0.375, 0.5  , 0.625, 0.75 , 0.875]],
+                     dtype=K.floatx()),
+        ),
+        (
+            6,
+            "smooth",
+            np.array(
+                [[-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75]],
+                dtype=K.floatx()),
+            np.array([[0.3125  , 0.359375, 0.40625 , 0.453125, 0.5     , 0.546875,
+            0.59375 , 0.640625]],
+                     dtype=K.floatx()),
+        ),
+        (
+            6,
+            "real",
+            np.array(
+                [[-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75]],
+                dtype=K.floatx()),
+            np.array([[0.265625, 0.328125, 0.375   , 0.4375  , 0.5     , 0.5625  ,
+            0.625   , 0.671875 ]],
+                     dtype=K.floatx()),
+        ),
+    ])
+def test_quantized_sigmoid(bits, sigmoid_type, test_values, expected_values):
+  """Test quantized_sigmoid function with three different sigmoid variants."""
+  qkeras.quantizers.set_internal_sigmoid(sigmoid_type)
+  x = K.placeholder(ndim=2)
+  f = K.function([x], [quantized_sigmoid(bits)(x)])
+  qkeras.quantizers.set_internal_sigmoid("hard")
```

Comment:
I don't think you can assume that the hard sigmoid is the default sigmoid. It is probably best to store the default sigmoid at the beginning of the test, and then re-set it.

Reply:
Hard sigmoid is set as `_sigmoid` here:

https://github.com/google/qkeras/blob/master/qkeras/quantizers.py#L204

It also is assumed as the sigmoid type in later `qrecurrent` tests.

Should I still change the behavior here?

Reply:
Sorry, let me clarify.

If we decide to change the default sigmoid later on, then we would change https://github.com/google/qkeras/blob/master/qkeras/quantizers.py#L204 to set `_sigmoid` to equal another kind of sigmoid function, such as `_sigmoid = smooth_sigmoid`. In this case, we would be changing the default sigmoid function used throughout qkeras. However, if we happen to run this test, the test will override our default sigmoid function and always set the internal sigmoid to `hard` after it finishes running, even when `hard` is not the default sigmoid we set in L204. 

This means if we were to change the default sigmoid function, we would need to look at all the tests to make sure no other code in qkeras changes the default sigmoid. To avoid this problem, we can instead change this test to "remember" what the default sigmoid function was set to, and then re-set it exactly as before when the test finishes running. 

This is one way of doing it:
```
previous_sigmoid = _sigmoid  # store the previous / default sigmoid
qkeras.quantizers.set_internal_sigmoid(sigmoid_type)
x = K.placeholder(ndim=2)
f = K.function([x], [quantized_sigmoid(bits)(x)])
_sigmoid = previous_sigmoid  # re-set the sigmoid function to what it was before
```

Reply:
Thanks for the comment. `_sigmoid` was out of scope but the now implemented way of storing the sigmoid works.